app/helpers/announcement_helper.rb

```diff
@@ -0,0 +1,29 @@
+module AnnouncementHelper
+  def format_published_at(params)
+    unless day_month_or_year_empty?(params)
```

Comment:
The double-negative hurting by brain a bit here.  I wonder if this method should be renamed to `full_date`?(params) and it could check for a truthy, i.e. `if full_date`?(params)`

Actually, if "year", "month" and "day" is used instead of params, the method becomes even simpler, as it could be changed to return: `year.present? && month.present? && day.present?`

Reply:
We reversed the logic to make this a bit easier to follow, in the interests of time we have not made the changes to the way we access the date data.